vnext/Microsoft.ReactNative/JsiApi.idl

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.ReactNative
+{
+  interface IJsiRuntime;
+
+  delegate void JsiDataHandler(UInt8[] data);
+
+  [webhosthidden]
+  interface IJsiBuffer
+  {
+    UInt32 Size { get; };
+    void GetData(JsiDataHandler handler);
+  };
+
+  enum JsiValueKind
+  {
+    Undefined,
+    Null,
+    Boolean,
+    Number,
+    Symbol,
+    String,
+    Object,
+  };
+
+  [webhosthidden]
+  struct JsiValueData
+  {
+    JsiValueKind Kind;
+    UInt64 Data;
+  };
+
+  struct JsiSymbolData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiStringData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiObjectData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiWeakObjectData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiPropertyNameIdData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiFunctionData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiArrayData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiArrayBufferData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiScopeState
+  {
+    UInt64 Data;
+  };
+
+  [default_interface]
+  runtimeclass JsiPreparedJavaScript
+  {
+  };
+
+  delegate JsiValueData JsiHostFunction(IJsiRuntime runtime, JsiValueData thisArg, JsiValueData[] args);
+
+  [webhosthidden]
+  interface IJsiHostObject
+  {
+    JsiValueData GetProperty(IJsiRuntime runtime, JsiPropertyNameIdData name);
+    void SetProperty(IJsiRuntime runtime, JsiPropertyNameIdData name, JsiValueData value);
+    IVector<JsiPropertyNameIdData> GetPropertyNames(IJsiRuntime runtime);
```

Comment:
>IVector [](start = 4, length = 7)

Where is this defined?

Reply:
It is a Windows::Foundation::Collection class. It is allowed to be used without the full namespace.

---
In reply to: [455234888](https://github.com/microsoft/react-native-windows/pull/5461#discussion_r455234888) [](ancestors = 455234888)